src/ripple/protocol/STArray.h

```diff
@@ -71,11 +71,35 @@ class STArray final
         return emplace(n, buf, std::move(*this));
     }
 
-    void push_back (const STObject & object)
+    STObject& operator[] (std::size_t j)
+    {
+        return v_[j];
+    }
+
+    STObject const& operator[] (std::size_t j) const
+    {
+        return v_[j];
+    }
+
+    STObject& back() { return v_.back(); }
+
+    template <class... Args>
+    void
+    emplace_back(Args&&... args)
+    {
+        v_.emplace_back(std::forward<Args>(args)...);
+    }
+
+    void push_back (STObject const& object)
     {
         v_.push_back(object);
     }
 
+    void push_back (STObject&& object)
+    {
```

Comment:
@HowardHinnant Can you please enlighten us on `push_back` versus `emplace_back` and when to use either one depending on the arguments?